### PR TITLE
tutorial: Knative Rust-Tide How To

### PR DESCRIPTION
**Pull request by No9**
## Checklist
- [x] commit message follows [commit guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md#commit-message-guidelines).
- [x] website builds and runs in production - for information on how to test locally go [here](https://github.com/appsody/website/blob/master/DEVELOPMENT.md).

## Related Issues
None

#issue-number
None

### CODE REVIEW

**Comment by neeraj-laad** [comment]
@CameronMcWilliam Would you be able to give this a first pass review?

File: content/tutorials/ServerlessRust.md [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+    Running command: docker run --rm --entrypoint /bin/bash experimental/appsody/rust-tide:0.1 -c "find /project -type f -name .appsody-init.sh"
+    Successfully added your project to /home/anton/.appsody/project.yaml
+    Your Appsody project ID has been set to 20200601194200.47030643
+    Successfully initialized Appsody project with the experimental/rust-tide stack and the default template.
+   ```
+
+   The most important lines to make note of are:
+
+   ```
+   Your Appsody project name has been set to `appsodyrusttide`
```
Comment (CameronMcWilliam) [inline content/tutorials/ServerlessRust.md:100]: really minor thing but appsodyrusttide was appsodyRustTide really really minor sorry! :)

File: content/tutorials/ServerlessRust.md [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+   ``` 
+   and
+   ```
+   Successfully initialized Appsody project with the nodejs-loopback stack and the default template.
```
Comment (CameronMcWilliam) [inline content/tutorials/ServerlessRust.md:104]: wrong stack

File: content/tutorials/ServerlessRust.md [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+   ```
+
+6. Now open this project in your favorite IDE.
+
+   This tutorial is going to use VS Code:
+
+   ```
+   code .
```
Comment (CameronMcWilliam) [inline content/tutorials/ServerlessRust.md:112]: don't think we need this VSCode bit just say open in your favourite IDE

File: content/tutorials/ServerlessRust.md [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+   ```
+
+7. The basic structure of the application looks like this:
+
+   ![rust_tide_appsody_template_files_1.png](./resources/rusttide/rust_tide_appsody_template_files_1.png)
+  
+   One important file is the Appsody configuration file for your project, which is named
+   `.appsody-config.yaml`. It defines the `name` of your project, and the stack on
```
Comment (CameronMcWilliam) [inline content/tutorials/ServerlessRust.md:120]: One important file is the Appsody configuration file for your project, `.appsody-config.yaml`: 

```yaml
    project-name: appsodyrusttide
    stack: experimental/rust-tide:0.1
    ```
It defines the `name` of your project, and the stack on  which it is based.

File: content/tutorials/ServerlessRust.md [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+    
+    ```rust
+    pub fn app() -> tide::Server<()> {    
+        let mut api = tide::new();
+        api.at("/").get(|_| async move { Ok("Hello, world!") });
+        api
+    }
+    ```
+
+    The initial application is very similar to the [example application](https://github.com/http-rs/tide/blob/master/examples/hello.rs), but
```
Comment (CameronMcWilliam) [inline content/tutorials/ServerlessRust.md:142]: Could put this in a > noteblock

File: content/tutorials/ServerlessRust.md [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+    it's a library that is [loaded by a server](https://github.com/No9/rust-tide/blob/master/image/project/server/bin/src/main.rs#L5) at runtime to reduce the amount of developer noise.
+
+In the next section, you are going modify and build this service.
```
Comment (CameronMcWilliam) [inline content/tutorials/ServerlessRust.md:145]: Move to the next section:

## Building Your Application

The template application is only a starting point, you will now modify and build this service.

File: content/tutorials/ServerlessRust.md [hunk 7]
```diff
@@ -0,0 +1,275 @@
...
+
+## Building Your Application
+
+The template application is only a starting point.
+
+Now you are going to modify the endpoint so it reads an environment variable and returns is as JSON.
```
Comment (CameronMcWilliam) [inline content/tutorials/ServerlessRust.md:151]: Firstly, you are going to modify....

Comment (CameronMcWilliam) [reply content/tutorials/ServerlessRust.md:151]: "... returns it as JSON"

File: content/tutorials/ServerlessRust.md [hunk 8]
```diff
@@ -0,0 +1,275 @@
...
+    it's a library that is [loaded by a server](https://github.com/No9/rust-tide/blob/master/image/project/server/bin/src/main.rs#L5) at runtime to reduce the amount of developer noise.
+
+In the next section, you are going modify and build this service.
+
+## Building Your Application
+
+The template application is only a starting point.
+
+Now you are going to modify the endpoint so it reads an environment variable and returns is as JSON. 
+This is based on the [tide json example](https://github.com/http-rs/tide/blob/master/examples/json.rs) but uses an environment variable rather than a post to demonstrate some of the Knative capabilities.
```
Comment (CameronMcWilliam) [inline content/tutorials/ServerlessRust.md:152]: "POST"

File: content/tutorials/ServerlessRust.md [hunk 9]
```diff
@@ -0,0 +1,275 @@
...
+
+1. The first thing to do is update the `Cargo.toml` to reference `serde` libraries. This gives our project JSON capaibilities.
```
Comment (CameronMcWilliam) [inline content/tutorials/ServerlessRust.md:154]: remove start of sentence:

"Update the `cargo.toml`...."

File: content/tutorials/ServerlessRust.md [hunk 10]
```diff
@@ -0,0 +1,275 @@
...
+    ```toml
+    [dependencies]
+    tide = { version = "0.9" }
+    serde = "1.0.102"
+    serde_json = "1.0.41"
+    ```
+
+2. Now we will update `lib.rs` to read the environment variable and return a Cat struct to the user.
```
Comment (CameronMcWilliam) [inline content/tutorials/ServerlessRust.md:162]: `Cat`

File: content/tutorials/ServerlessRust.md [hunk 11]
```diff
@@ -0,0 +1,275 @@
...
+                let cat = Cat {
+                    name: env_my_name.into(),
+                };
+                Ok(Response::new(StatusCode::Ok).body_json(&cat)?)
+            });
+            api
+        }
+    ```
+
+3. Once the code changes are complete you should be able to run the service with:
```
Comment (CameronMcWilliam) [inline content/tutorials/ServerlessRust.md:187]: "...you can run the service with:"

File: content/tutorials/ServerlessRust.md [hunk 12]
```diff
@@ -0,0 +1,275 @@
...
+            });
+            api
+        }
+    ```
+
+3. Once the code changes are complete you should be able to run the service with: 
+    ```
+    appsody run --docker-options "--env MY_NAME=tibbs"
+    ```
+    where `--docker-options` passes the environment settings to the container
```
Comment (CameronMcWilliam) [inline content/tutorials/ServerlessRust.md:191]: remove where
put in noteblock

File: content/tutorials/ServerlessRust.md [hunk 13]
```diff
@@ -0,0 +1,275 @@
...
+
+4. The service should now be available on http://localhost:8000/ and if you open it in a browser it will now return.
```
Comment (CameronMcWilliam) [inline content/tutorials/ServerlessRust.md:193]: "The service will be available at http://localhost:8000/ and accessing it in a browser will return:"

File: content/tutorials/ServerlessRust.md [hunk 14]
```diff
@@ -0,0 +1,275 @@
...
+    ```json
+        {
+            "name" : "tibbs"
+        }
+    ```
+
+5. Now we need to create a production build for the application. Make user you are logged into [docker hub](https://docs.docker.com/engine/reference/commandline/login/) before running it.
```
Comment (CameronMcWilliam) [inline content/tutorials/ServerlessRust.md:200]: "Make sure you are logged into [Docker Hub](https://docs.docker.com/engine/reference/commandline/login/) and run:

File: content/tutorials/ServerlessRust.md [hunk 15]
```diff
@@ -0,0 +1,275 @@
...
+
+    ```
+    $ appsody build -t YOUR_DOCKER_USER/appsodyrusttide:v1.0.0 --publish --knative
```
Comment (CameronMcWilliam) [inline content/tutorials/ServerlessRust.md:203]: again make sure that appsodyrusttide is consistent with what you decide earlier (`appsodyRustTide`)

File: content/tutorials/ServerlessRust.md [hunk 16]
```diff
@@ -0,0 +1,275 @@
...
+    ```
+
+Now we are ready to deploy your application
```
Comment (CameronMcWilliam) [inline content/tutorials/ServerlessRust.md:206]: Don't think we need this

File: content/tutorials/ServerlessRust.md [hunk 17]
```diff
@@ -0,0 +1,275 @@
...
+
+## Deploy Your Application
+
+This section will be split into 3 sections. The first will cover deploying to Knative on IBM Cloud. The second will cover a plain Knative deploy and the final one will cover a preconfigured OpenShift deployment
```
Comment (CameronMcWilliam) [inline content/tutorials/ServerlessRust.md:210]: This section will be split into three sub-sections:
1.  Deploy to Knative on IBM Cloud
2. Deploy to Minikube
3. Deploy to Openshift

File: content/tutorials/ServerlessRust.md [hunk 18]
```diff
@@ -0,0 +1,275 @@
...
+
+### Deploy the image to IBM Cloud
```
Comment (CameronMcWilliam) [inline content/tutorials/ServerlessRust.md:212]: ### Deploy to Knative on IBM Cloud

File: content/tutorials/ServerlessRust.md [hunk 19]
```diff
@@ -0,0 +1,275 @@
...
+
+Now we are ready to deploy your application
+
+## Deploy Your Application
+
+This section will be split into 3 sections. The first will cover deploying to Knative on IBM Cloud. The second will cover a plain Knative deploy and the final one will cover a preconfigured OpenShift deployment
+
+### Deploy the image to IBM Cloud 
+
+    Ensure that the you have created a project for the service following [these instructions](https://cloud.ibm.com/docs/knative?topic=knative-manage-project#create-project-cli).
```
Comment (CameronMcWilliam) [inline content/tutorials/ServerlessRust.md:214]: "Ensure that you have [created a project for the service](https://cloud.ibm.com/docs/knative?topic=knative-manage-project#create-project-cli), then run the following command:"

File: content/tutorials/ServerlessRust.md [hunk 20]
```diff
@@ -0,0 +1,275 @@
...
+
+### Deploy the image to IBM Cloud 
+
+    Ensure that the you have created a project for the service following [these instructions](https://cloud.ibm.com/docs/knative?topic=knative-manage-project#create-project-cli). 
+
+    ```
+    $ ibmcloud coligo application create --name appsodyrusttide --image YOUR_DOCKER_USER/appsodyrusttide:v1.0.0 --env MY_NAME=tibbs
+    ```
+
+### Deploy the image to minikube
```
Comment (CameronMcWilliam) [inline content/tutorials/ServerlessRust.md:220]: ### Deploy to Minikube

File: content/tutorials/ServerlessRust.md [hunk 21]
```diff
@@ -0,0 +1,275 @@
...
+    template:
+        spec:
+        containers:
+            - image: docker.io/YOUR_DOCKER_USER/appsodyrusttide:v1.0.0 # The URL to the image of the app
+              env:
+                - name: MY_NAME # The environment variable printed out by the sample app
+                  value: "tibbs"
+    ```
+
+2. Push the service
```
Comment (CameronMcWilliam) [inline content/tutorials/ServerlessRust.md:240]: Colon at the end

File: content/tutorials/ServerlessRust.md [hunk 22]
```diff
@@ -0,0 +1,275 @@
...
+
+    Ensure that the you have created a project for the service following [these instructions](https://cloud.ibm.com/docs/knative?topic=knative-manage-project#create-project-cli). 
+
+    ```
+    $ ibmcloud coligo application create --name appsodyrusttide --image YOUR_DOCKER_USER/appsodyrusttide:v1.0.0 --env MY_NAME=tibbs
+    ```
+
+### Deploy the image to minikube 
+    
+1. Create a service definition in the application folder
```
Comment (CameronMcWilliam) [inline content/tutorials/ServerlessRust.md:222]: Colon at the end

File: content/tutorials/ServerlessRust.md [hunk 23]
```diff
@@ -0,0 +1,275 @@
...
+    
+    ```yaml
+    apiVersion: serving.knative.dev/v1 # Current version of Knative
+    kind: Service
+    metadata:
+        name: appsodyrusttide # The name of the app
```
Comment (CameronMcWilliam) [inline content/tutorials/ServerlessRust.md:228]: appsodyRustTide or appsodyrusttide, same comments as before

File: content/tutorials/ServerlessRust.md [hunk 24]
```diff
@@ -0,0 +1,275 @@
...
+## Deploy Your Application
+
+This section will be split into 3 sections. The first will cover deploying to Knative on IBM Cloud. The second will cover a plain Knative deploy and the final one will cover a preconfigured OpenShift deployment
+
+### Deploy the image to IBM Cloud 
+
+    Ensure that the you have created a project for the service following [these instructions](https://cloud.ibm.com/docs/knative?topic=knative-manage-project#create-project-cli). 
+
+    ```
+    $ ibmcloud coligo application create --name appsodyrusttide --image YOUR_DOCKER_USER/appsodyrusttide:v1.0.0 --env MY_NAME=tibbs
```
Comment (CameronMcWilliam) [inline content/tutorials/ServerlessRust.md:217]: appsodyRustTide or appsodyrusttide

File: content/tutorials/ServerlessRust.md [hunk 25]
```diff
@@ -0,0 +1,275 @@
...
+    apiVersion: serving.knative.dev/v1 # Current version of Knative
+    kind: Service
+    metadata:
+        name: appsodyrusttide # The name of the app
+        namespace: default # The namespace the app will use
+    spec:
+    template:
+        spec:
+        containers:
+            - image: docker.io/YOUR_DOCKER_USER/appsodyrusttide:v1.0.0 # The URL to the image of the app
```
Comment (CameronMcWilliam) [inline content/tutorials/ServerlessRust.md:234]: appsodyrusttide or appsodyRustTide

File: content/tutorials/ServerlessRust.md [hunk 26]
```diff
@@ -0,0 +1,275 @@
...
+
+1. Deploying to OpenShift is very straight forward and just requires: 
+
+    ```
+    $ appsody deploy --no-build --knative 
+    ```
+
+## Validate Your Application
+
+1. Using the kn cli get the list of services
```
Comment (CameronMcWilliam) [inline content/tutorials/ServerlessRust.md:255]: 1. Get the list of services using the kn cli:

```
     $ kn service list
     
     NAME              URL                                                                     LATEST                     AGE    CONDITIONS   READY   REASON
     appsodyrusttide   http://appsodyrusttide.604053c7-2682.us-south.knative.appdomain.cloud   appsodyrusttide-s4lwr5-3   2d7h   3 OK / 3     True    
     ```

File: content/tutorials/ServerlessRust.md [hunk 27]
```diff
@@ -0,0 +1,275 @@
...
+
+    ```
+    $ kn service list
+    
+    NAME              URL                                                                     LATEST                     AGE    CONDITIONS   READY   REASON
+    appsodyrusttide   http://appsodyrusttide.604053c7-2682.us-south.knative.appdomain.cloud   appsodyrusttide-s4lwr5-3   2d7h   3 OK / 3     True
```
Comment (CameronMcWilliam) [inline content/tutorials/ServerlessRust.md:261]: appsodyRustTide or appsodyrusttide

File: content/tutorials/ServerlessRust.md [hunk 28]
```diff
@@ -0,0 +1,275 @@
...
+    $ kn service list
+    
+    NAME              URL                                                                     LATEST                     AGE    CONDITIONS   READY   REASON
+    appsodyrusttide   http://appsodyrusttide.604053c7-2682.us-south.knative.appdomain.cloud   appsodyrusttide-s4lwr5-3   2d7h   3 OK / 3     True    
+    ```
+
+2. Test the URL returned in the list.
+
+    ```
+    $ curl http://appsodyrusttide.604053c7-2682.us-south.knative.appdomain.cloud
```
Comment (CameronMcWilliam) [inline content/tutorials/ServerlessRust.md:267]: appsodyRustTide or ... Im sure I don't need to say it again haha :)

File: content/tutorials/ServerlessRust.md [hunk 29]
```diff
@@ -0,0 +1,275 @@
...
+    ```
+    $ curl http://appsodyrusttide.604053c7-2682.us-south.knative.appdomain.cloud 
+    {
+      "name":"tibbs"  
+    }
+    ```
+
+## Summary
+
+This tutorial was a starter to get a simple service up and running. In future there will be examples of debugging and connecting to backend systems such as databases. In the mean time checkout some of the other appsody [tutorials for other stacks](https://appsody.dev/tutorials/).
```
Comment (CameronMcWilliam) [inline content/tutorials/ServerlessRust.md:275]: "In the future...."

Comment (CameronMcWilliam) [reply content/tutorials/ServerlessRust.md:275]: Last sentence delete I think.

**Comment by neeraj-laad** [comment]
@No9 Are you able to address these review comment so we can merge this?

**Comment by No9** [comment]
Hey @neeraj-laad yes I intend to look at it over the weekend.

File: content/tutorials/ServerlessRust.md [hunk 30]
```diff
@@ -0,0 +1,275 @@
...
+tags:
+  - programming
+  - serverless
+  - knative
+  - tide
+  - rust
+  - appsody
+cover: "https://venshare.com/images/tide-appsody-kn.png"
+---
+
```
[the remark this answers is not in the document]
Comment (No9) [reply content/tutorials/ServerlessRust.md:19]: On line 2 set the title
```
title: "Serverless Rust with Tide, Appsody and Knative"
```
Or is there another place i need to do it?

File: content/tutorials/ServerlessRust.md [hunk 31]
```diff
@@ -0,0 +1,275 @@
...
+## Introduction
+
+Following the feedback from an [Appsody on OpenShift](https://us02web.zoom.us/webinar/register/rec/WN__gBLTiqLQPiDRUkzdWWXZg?meetingId=2udFNa2s0HNOQ4nnz2LRW7A4A9jqX6a81nUe__Vcz0uAmF0Mz4Yqwn_obRPTpJuz&playId=usctfrr7_zw3GoGQ5gSDC_d6W9W5K_6s13RMrKdcnUa2AiVSN1ekbrREMOrwm00F71LWvXjt14jaBZaf&action=play&_x_zm_rtaid=5uZDfWV9RYy-J3EBnxrctQ.1590833520420.dc63c5dacd2e261a888c87d88a17324b&_x_zm_rhtaid=145) presentation that I did recently I thought I would continue on the theme and write a step by step tutorial on how to scaffold and build a serverless rust application with Appsody and deploy it into a knative environment.
+
+## Prerequisites
+
+In order to follow this tutorial, you need to install the Appsody CLI, which also requires you to have Docker installed. You will also need a Knative environment to deploy to.
```
[the remark this answers is not in the document]
Comment (No9) [reply content/tutorials/ServerlessRust.md:26]: Agreed - I've removed and added a link to the docker install

File: content/tutorials/ServerlessRust.md [hunk 32]
```diff
@@ -0,0 +1,275 @@
...
+   mkdir appsodyRustTide
+   ```
+
+3. Navigate to that directory:
+
+   ```
+   cd appsodyRustTide
+   ```
+
+4. List the available Appsody stacks (and templates) with:
```
[the remark this answers is not in the document]
Comment (No9) [reply content/tutorials/ServerlessRust.md:58]: Agreed

File: content/tutorials/ServerlessRust.md [hunk 33]
```diff
@@ -0,0 +1,275 @@
...
+
+### Create a Knative Environment
+
+- [Fast - Knative on IBM Cloud](https://cloud.ibm.com/docs/knative?topic=knative-kn-install-cli)
+
+- [Local - Minikube](https://knative.dev/v0.12-docs/install/knative-with-minikube/)
+
+- [Anywhere - OpenShift](https://docs.openshift.com/container-platform/4.3/serverless/installing_serverless/installing-knative-serving.html)
+
+You should also be able to use the appsody build in any Knative environment if you already have one with your favourite cloud provider.
```
[the remark this answers is not in the document]
Comment (No9) [reply content/tutorials/ServerlessRust.md:40]: Updated

File: content/tutorials/ServerlessRust.md (hunk 2 above)
Comment (No9) [reply content/tutorials/ServerlessRust.md:104]: Updated

File: content/tutorials/ServerlessRust.md (hunk 1 above)
Comment (No9) [reply content/tutorials/ServerlessRust.md:100]: Appsody removes the capitalization in the generation and command output hence the mismatch. 
I can see why that is jarring so i've updated the instructions to just use appsodyrusttide

File: content/tutorials/ServerlessRust.md (hunk 3 above)
Comment (No9) [reply content/tutorials/ServerlessRust.md:112]: I've removed it the VS code reference but kept the step.

File: content/tutorials/ServerlessRust.md (hunk 4 above)
Comment (No9) [reply content/tutorials/ServerlessRust.md:120]: Updated

File: content/tutorials/ServerlessRust.md (hunk 5 above)
Comment (No9) [reply content/tutorials/ServerlessRust.md:142]: Not sure what you are driving at here with a note block
For steps 7.8.9 or just 9? 
Also don't want to loose the formatting of the code.

File: content/tutorials/ServerlessRust.md (hunk 7 above)
Comment (No9) [reply content/tutorials/ServerlessRust.md:151]: Applied your suggestion and reworded

File: content/tutorials/ServerlessRust.md (hunk 6 above)
Comment (No9) [reply content/tutorials/ServerlessRust.md:145]: Updated

File: content/tutorials/ServerlessRust.md (hunk 8 above)
Comment (No9) [reply content/tutorials/ServerlessRust.md:152]: Updated

File: content/tutorials/ServerlessRust.md (hunk 9 above)
Comment (No9) [reply content/tutorials/ServerlessRust.md:154]: Updated

File: content/tutorials/ServerlessRust.md (hunk 10 above)
Comment (No9) [reply content/tutorials/ServerlessRust.md:162]: Updated

File: content/tutorials/ServerlessRust.md (hunk 11 above)
Comment (No9) [reply content/tutorials/ServerlessRust.md:187]: Updated

File: content/tutorials/ServerlessRust.md (hunk 12 above)
Comment (No9) [reply content/tutorials/ServerlessRust.md:191]: Updated

File: content/tutorials/ServerlessRust.md (hunk 13 above)
Comment (No9) [reply content/tutorials/ServerlessRust.md:193]: Updated

File: content/tutorials/ServerlessRust.md (hunk 14 above)
Comment (No9) [reply content/tutorials/ServerlessRust.md:200]: Updated

File: content/tutorials/ServerlessRust.md (hunk 15 above)
Comment (No9) [reply content/tutorials/ServerlessRust.md:203]: Updated

File: content/tutorials/ServerlessRust.md (hunk 16 above)
Comment (No9) [reply content/tutorials/ServerlessRust.md:206]: Removed

File: content/tutorials/ServerlessRust.md (hunk 17 above)
Comment (No9) [reply content/tutorials/ServerlessRust.md:210]: Updated

File: content/tutorials/ServerlessRust.md (hunk 18 above)
Comment (No9) [reply content/tutorials/ServerlessRust.md:212]: Updated

File: content/tutorials/ServerlessRust.md (hunk 19 above)
Comment (No9) [reply content/tutorials/ServerlessRust.md:214]: Updated

File: content/tutorials/ServerlessRust.md (hunk 24 above)
Comment (No9) [reply content/tutorials/ServerlessRust.md:217]: appsodyrusttide

File: content/tutorials/ServerlessRust.md (hunk 20 above)
Comment (No9) [reply content/tutorials/ServerlessRust.md:220]: Updated

File: content/tutorials/ServerlessRust.md (hunk 23 above)
Comment (No9) [reply content/tutorials/ServerlessRust.md:228]: appsodyrusttide

File: content/tutorials/ServerlessRust.md (hunk 25 above)
Comment (No9) [reply content/tutorials/ServerlessRust.md:234]: appsodyrusttide

File: content/tutorials/ServerlessRust.md (hunk 26 above)
Comment (No9) [reply content/tutorials/ServerlessRust.md:255]: Updated

File: content/tutorials/ServerlessRust.md (hunk 27 above)
Comment (No9) [reply content/tutorials/ServerlessRust.md:261]: appsodyrusttide

File: content/tutorials/ServerlessRust.md (hunk 28 above)
Comment (No9) [reply content/tutorials/ServerlessRust.md:267]: appsodyrusttide

File: content/tutorials/ServerlessRust.md (hunk 29 above)
Comment (No9) [reply content/tutorials/ServerlessRust.md:275]: Updated

File: content/tutorials/ServerlessRust.md [hunk 34]
```diff
@@ -0,0 +1,275 @@
...
+## Prerequisites
+
+In order to follow this tutorial, you need to install the Appsody CLI, which also requires you to have Docker installed. You will also need a Knative environment to deploy to.
+
+### Install the Appsody CLI
+
+Follow the [Installing Appsody](https://appsody.dev/docs/getting-started/installation) guide to install the CLI for your platform.
+
+### Create a Knative Environment
+
```
[the remark this answers is not in the document]
Comment (No9) [reply content/tutorials/ServerlessRust.md:35]: Updated

**Comment by No9** [comment]
Hey @CameronMcWilliam 
I've applied all the recommendations just these 2 outstanding items I think:

1. Clarification on the adding a `title`  I think it's already there

2. What exactly you would recommend to turn into a note around line 128

Thanks for such a thorough review

File: content/tutorials/ServerlessRust.md (hunk 30 above)
[the remark this answers is not in the document]
Comment (CameronMcWilliam) [reply content/tutorials/ServerlessRust.md:19]: That's good there, also put it in a # at the top too

#Serverless Rust with Tide, Appsody and Knative

File: content/tutorials/ServerlessRust.md (hunk 5 above)
Comment (CameronMcWilliam) [reply content/tutorials/ServerlessRust.md:141]: The initial application.... Could be in a note block.  But It doesn't have to be if it looks weird!

Comment (No9) [reply content/tutorials/ServerlessRust.md:141]: Ah I see what you mean - It improves it so I have gone with your suggestion.

**Comment by No9** [comment]
@CameronMcWilliam Thanks for those clarifications - I have applied them and updated the PR.

**Comment by CameronMcWilliam** [comment]
> @CameronMcWilliam Thanks for those clarifications - I have applied them and updated the PR.

I'll just run it locally one more time on my machine so I can see what It looks like on the website and approve it

**Pull request merged**